osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/test/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/misc/GetGsmDiagnosticCommandExecutorIntegrationTest.java

```diff
@@ -46,51 +50,157 @@
 import org.opensmartgridplatform.shared.infra.jms.MessageMetadata;
 
 @ExtendWith(MockitoExtension.class)
-public class GetGsmDiagnosticCommandExecutorIntegrationTest {
+class GetGsmDiagnosticCommandExecutorIntegrationTest {
 
   private GetGsmDiagnosticCommandExecutor executor;
 
   private DlmsConnectionManagerStub connectionManagerStub;
   private DlmsConnectionStub connectionStub;
 
+  private ObjectConfigService objectConfigService;
+
   @BeforeEach
-  public void setUp() {
+  public void setUp() throws IOException, ObjectConfigException {
+    this.objectConfigService = new ObjectConfigService(null);
+
     final DlmsHelper dlmsHelper = new DlmsHelper();
     final DlmsObjectConfigConfiguration dlmsObjectConfigConfiguration =
         new DlmsObjectConfigConfiguration();
-    final DlmsObjectConfigService dlmsObjectConfigService =
-        new DlmsObjectConfigService(
-            dlmsHelper, dlmsObjectConfigConfiguration.getDlmsObjectConfigs());
 
-    this.executor = new GetGsmDiagnosticCommandExecutor(dlmsHelper, dlmsObjectConfigService);
+    this.executor = new GetGsmDiagnosticCommandExecutor(dlmsHelper, this.objectConfigService);
+
     this.connectionStub = new DlmsConnectionStub();
     this.connectionManagerStub = new DlmsConnectionManagerStub(this.connectionStub);
-
     this.connectionStub.setDefaultReturnValue(DataObject.newArrayData(Collections.emptyList()));
   }
 
-  @Test
-  public void testExecuteDsmr4() throws Exception {
-    for (final CommunicationMethod method : CommunicationMethod.values()) {
-      this.testExecute(Protocol.DSMR_4_2_2, method, true);
-    }
+  @ParameterizedTest
+  @CsvSource({
+    "DSMR_4_2_2,GPRS",
+    "SMR_4_3,CDMA",
+    "SMR_5_0_0,GPRS",
+    "SMR_5_0_0,CDMA",
+    "SMR_5_1,CDMA",
+    "SMR_5_1,GPRS",
+    "SMR_5_2,CDMA",
+    "SMR_5_2,GPRS",
+    "SMR_5_2,LTE",
+    "SMR_5_5,LTE",
+  })
+  void executeAndValidateValids(final String protocol, final String communicationMethod)
+      throws Exception {
+
+    this.executeAndValidate(
+        Protocol.valueOf(protocol), CommunicationMethod.valueOf(communicationMethod), false);
   }
 
-  @Test
-  public void testExecuteSmr5_0() throws Exception {
-    for (final CommunicationMethod method : CommunicationMethod.values()) {
-      this.testExecute(Protocol.SMR_5_0_0, method, true);
-    }
+  @ParameterizedTest
+  @CsvSource({
+    "DSMR_4_2_2,GPRS",
+    "SMR_4_3,CDMA",
+    "SMR_4_3,LTE",
+    "SMR_5_0_0,LTE",
+    "SMR_5_1,LTE",
+    "SMR_5_2,CDMA",
+    "SMR_5_5,GPRS",
+    "SMR_5_5,CDMA"
+  })
+  void executeAndValidateInvalids(final String protocol, final String communicationMethod)
+      throws Exception {
+
+    this.executeAndValidate(
+        Protocol.valueOf(protocol), CommunicationMethod.valueOf(communicationMethod), true);
   }
```

Comment:
These tests suggest to test two exclusive sets. But they are not. There is some overlap in the testsets and almost every combination of protocol version and variant seems to be successful in both tests except a few which raises the question what this test actually does test.

Reply:
If the tests are supposed to be excluding it would be more clear to merge these and add the _expectObjectNotFound_ boolean to the CsvSource

Reply:
The exception assertions is not correct, I will correct this.